rpcs3/Emu/Io/usb_device.cpp

```diff
@@ -200,24 +200,20 @@ bool usb_device_emulated::open_device()
 
 u32 usb_device_emulated::get_descriptor(u8 type, u8 index, u8* buf, u32 buf_size)
 {
-	u32 expected_count = 2;
+	const std::array<u8, 2> header {header.size(), type};
```

Comment:
size() is not static, isn't that a UB or may return unspecified value? Im not sure what the c++ rules are for this.

Reply:
Since it's std::array instead of the other STL containers like vector or map, its size() is constexpr according to https://en.cppreference.com/w/cpp/container/array/size

Reply:
Yeah but this rule applies here? not every constexpr function may actually be constexpr at all times, this is what consteval is for. Im asking because Im not sure it's a good idea to call non-statc methods before an object is initialized.

Reply:
Would this be any better?
```C++
std::array<u8, 2> header;
header = {header.size(), type};
```

Reply:
Yeah

Reply:
Updated.